test/printer.js

```diff
@@ -1605,4 +1605,41 @@ describe("printer", function() {
         assert.strictEqual(pretty, code);
       });
     });
+
+    it("adds parenthesis around object expression", function() {
+        var code = "({}).x = 1;";
+
+        var assignment = b.assignmentExpression(
+            '=',
+            b.memberExpression(b.objectExpression([]), b.identifier('x'), false),
+            b.literal(1)
+        );
+
+        var ast = b.program([
+            b.expressionStatement(assignment)
+        ]);
+
+        var printer = new Printer({
+            arrowParensAlways: true
+        });
+        var pretty = printer.printGenerically(ast).code;
+        assert.strictEqual(pretty, code);
+    });
+
+    it("adds parenthesis around conditional", function() {
+        var code = 'new ((typeof a ? b : c))();';
```

Comment:
Why the extra parentheses here? Is this actually desirable, or are we erring on the side of safety (and this is just an edge case)?